### PR TITLE
Fix incorrect kvlds status values

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: tests/kvlds/main.c [hunk 1]
```diff
@@ -183,11 +183,11 @@ add(struct wire_requestqueue * Q,
 	}
 
 	/* Check for no-op. */
-	if (noop && op_p) {
+	if (noop && !op_p) {
```
Comment (gperciva) [inline tests/kvlds/main.c]: Instead of changing all these values, you might prefer it if we negated `done` when setting `op_p` on line 69.

Comment (cperciva) [reply tests/kvlds/main.c]: Yes.  The value `op_p` means "did we perform an operation".  The function `callback_donep` should take a final parameter of `int status` and set `op_p = status ? 0 : 1`.

File: kvlds/dispatch_mr.c [hunk 2]
```diff
@@ -524,7 +524,7 @@ callback_synced(void * cookie)
 			break;
 		case PROTO_KVLDS_CAD:
 			if (proto_kvlds_response_cad(B->WQ, R->ID,
-			    req->opdone))
+			    !req->opdone))
```
Comment (cperciva) [inline kvlds/dispatch_mr.c]: I would spell these as `req->opdone ? 0 : 1` to make it clear that we want those two particular values and not merely logical negation.

**Comment by gperciva** [comment]
Updated with a full rebase, because the new code is easier to understand without all the lines undoing the previous changes.

**Pull request merged**